src/main/java/eight/controller/Application.java

```diff
@@ -0,0 +1,82 @@
+package eight.controller;
+
+import com.github.javafaker.Faker;
+import eight.exception.NoSuchGenreException;
+import eight.exception.NoSuchTitleException;
+import eight.exception.NotAYearException;
+import eight.exception.UnsupportedDurationException;
+import eight.model.FilmCategory;
+import eight.model.Movie;
+
+import java.time.LocalTime;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Locale;
+import java.util.Random;
+import java.util.stream.Collectors;
+
+public class Application {
+
+    private List<Movie> movies;
+
+    public Application() {
+        movies = new ArrayList<>();
+        Faker faker = new Faker(new Locale("en-US"));
+        Random random = new Random();
+
+        for (int i = 0; i < 10; i++) {
+            movies.add(new Movie(LocalTime.of(random.nextInt(24), random.nextInt(59)),
+                    faker.name().title(), FilmCategory.getRandomCategory(), (1.0 + (3.0 - 1.0) * random.nextDouble()),
+                    random.nextInt(2020 - 1990) + 1990));
+        }
+    }
+
+    public void showSessionList() {
+        movies.forEach(movie -> System.out.println(movie.getSession() + "\t" + movie.getTitle() + "\t" +
+                movie.getGenre() + "\t" + movie.getDuration() + "\t" + movie.getYear()));
+    }
+
+    public void filterByGenre(String genre) throws NoSuchGenreException {
+        if (genre.matches("\\d*")) {
+            throw new NoSuchGenreException("Not a genre format");
+        }
+
+        movies.stream().filter(movie -> movie.getGenre().equalsIgnoreCase(genre))
+                .forEach(movie -> System.out.println(movie.getSession() + "\t" + movie.getTitle()));
+    }
+
+    public void sortMovies() {
+        movies.stream().map(Movie::getTitle).sorted().forEach(System.out::println);
+    }
+
+    public boolean isAtLeastOneMovie(double duration) throws UnsupportedDurationException {
```

Comment:
poor method name by its logic it would be better to use hasDurationMoreThen(double duration)

Reply:
Fixed